lib/explorer/backend/lazy_series.ex

```diff
@@ -635,6 +661,51 @@ defmodule Explorer.Backend.LazySeries do
   defp resolve_numeric_dtype(:window_mean, _items), do: :float
   defp resolve_numeric_dtype(_op, items), do: resolve_numeric_dtype(items)
 
+  defp resolve_numeric_temporal_dtype(op, %Series{dtype: ldt} = left, %Series{dtype: rdt} = right) do
+    case {op, ldt, rdt} do
+      {:add, {:datetime, ltu}, {:duration, rtu}} -> {:datetime, highest_precision(ltu, rtu)}
+      {:add, {:duration, ltu}, {:datetime, rtu}} -> {:datetime, highest_precision(ltu, rtu)}
+      {:add, {:duration, ltu}, {:duration, rtu}} -> {:duration, highest_precision(ltu, rtu)}
+      {:subtract, {:datetime, ltu}, {:datetime, rtu}} -> {:duration, highest_precision(ltu, rtu)}
+      {:subtract, {:datetime, ltu}, {:duration, rtu}} -> {:datetime, highest_precision(ltu, rtu)}
+      {:subtract, {:duration, ltu}, {:duration, rtu}} -> {:duration, highest_precision(ltu, rtu)}
+      {:multiply, :integer, {:duration, tu}} -> {:duration, tu}
+      {:multiply, {:duration, tu}, :integer} -> {:duration, tu}
+      {:divide, {:duration, tu}, :integer} -> {:duration, tu}
```

Comment:
Should we raise for `{:divide, :integer, {:duration, _}}`?

Reply:
Actually, I can see in series.ex that we do support `integer / duration` and we cast it into a duration? But I wonder if we should raise instead.

Reply:
I think we should raise because `integer / duration` would have the wrong units. `duration / integer` can still be measured in seconds, but `integer / duration` would be measured in Hertz.

Reply:
:+1: on raising then. But we would raise it on Explorer.Series, so not a concern here. :)

Reply:
Addressed by:
* [3357f90](https://github.com/elixir-explorer/explorer/pull/683/commits/3357f90078de9187bafc18499df89a3682e8d020)
* [743b306](https://github.com/elixir-explorer/explorer/pull/683/commits/743b3063512e1e8813c6ec01f48cb77ccbc29714)